server/src/main/java/org/elasticsearch/search/aggregations/support/MultiValuesSourceFieldConfig.java

```diff
@@ -78,7 +79,7 @@ private MultiValuesSourceFieldConfig(String fieldName, Object missing, Script sc
     }
 
     public MultiValuesSourceFieldConfig(StreamInput in) throws IOException {
-        this.fieldName = in.readString();
+        this.fieldName = in.readOptionalString();
```

Comment:
I think that if we want to change this, we might need to wrap this in version checks?  `readOptionalString()` encodes an extra byte for the boolean

Reply:
good catch. yes.

Reply:
OK. maybe we can do this in a separate PR. I think Docs need clean up too. We claim that `field` is required, but it can be replaced with `script`.

I see this as a bug, but want to attack this separately since the original issue was only with XContent.

Reply:
++ makes sense to me